pkg/provider/redis.go

```diff
@@ -0,0 +1,146 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+	"github.com/slack-go/slack"
+	"go.uber.org/zap"
+)
+
+
+type RedisClient struct {
+	client *redis.Client
+	logger *zap.Logger
+}
+
+func NewRedisClient(logger *zap.Logger) (*RedisClient, error) {
```

Comment:
wondering if we can initialize the client with the `teamID` instead of passing it through in all the fn calls